aten/src/ATen/core/Capsule.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <c10/util/intrusive_ptr.h>
+
+struct Capsule final: public c10::intrusive_ptr_target  {
+  /**
+   * Initializes an empty Capsule.
+   */
+  void* ptr;
+  Capsule(): ptr(nullptr) {}
+  Capsule(void* ptr_): ptr(ptr_) {}
+  ~Capsule() {
```

Comment:
I don't see a place where we're freeing or destroying the contained pointer. How do we ensure that the memory is cleaned up?

Reply:
We currently... don't. This needs to be resolved, but it's tricky to do so. This was what the whole conversation about forcing users to inherit from `intrusive_ptr` and such were about.